/*
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Library General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * ArtNetConfigMessages.proto
 * Defines the protocol buffers used to configure the Artnet device
 * Copyright (C) 2005 Simon Newton
 */

package ola.plugin.artnet;

// This adjusts the ArtNet settings for this node.
message OptionsRequest {
  optional string short_name = 1;
  optional string long_name = 2;
  optional int32 subnet = 3;
  optional int32 net = 4;
}

message OptionsReply {
  required uint32 status = 1;
  required string short_name = 2;
  required string long_name = 3;
  required int32 subnet = 4;
  required int32 net = 5;
}

// This returns a list of nodes configured to receive on a particular universe.
message NodeListRequest {
  required int32 universe = 1;
}

message OutputNode {
  required uint32 ip_address = 1;
}

// The list of output nodes
message NodeListReply {
  repeated OutputNode node = 1;
}

message Request {
  enum RequestType {
    ARTNET_OPTIONS_REQUEST = 1;
    ARTNET_NODE_LIST_REQUEST = 2;
  }

  required RequestType type = 1;
  optional OptionsRequest options = 2;
  optional NodeListRequest node_list = 3;
}

message Reply {
  enum ReplyType {
    ARTNET_OPTIONS_REPLY = 1;
    ARTNET_NODE_LIST_REPLY = 2;
  }
  required ReplyType type = 1;

  optional OptionsReply options = 2;
  optional NodeListReply node_list = 3;
}
